### PR TITLE
add doc on how to contribute

### PR DESCRIPTION
**Pull request by bvishal8510**
solve issue #198

### CODE REVIEW

**Comment by nathfreder** [comment]
This seems a bit "exact" and does not look very appealing.

File: how_to_contribute.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+7. It is judged based on some test written by repository holder
+8. Your code changes should withstand those tests
+9. If you pass test and have no conflict with master then it will show mergable
+10. You can click on open pull request button to open PR and contribute
+11. when you click, it will ask for a message
+12. In that mention the issue number for eg. "solved issue #198"
+13. finally click on open pull request
+14. Either repository holder will directly pull the changes or will tell you to do some changes
+15. You will need to modify your code until it is completely correct.
+16. Atlast your pull request will be merged
```
Comment (nathfreder) [inline how_to_contribute.md]: "Atlast" should be "At last"

File: how_to_contribute.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+4. Never make changes on master directly
+5. Do the necessary changes needed in the code to solve the problem
+6. push it on the github and as you push it will show you that branch is mergable or not
+7. It is judged based on some test written by repository holder
+8. Your code changes should withstand those tests
+9. If you pass test and have no conflict with master then it will show mergable
+10. You can click on open pull request button to open PR and contribute
+11. when you click, it will ask for a message
+12. In that mention the issue number for eg. "solved issue #198"
+13. finally click on open pull request
```
Comment (nathfreder) [inline how_to_contribute.md]: "open pull request" should be in double quotes.

File: how_to_contribute.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+1. Browse the Alex repository
+2. Have a look at the issues and if you are interested in any issue then clone the repository
+3. Always make your own branch on their repository
+4. Never make changes on master directly
+5. Do the necessary changes needed in the code to solve the problem
+6. push it on the github and as you push it will show you that branch is mergable or not
+7. It is judged based on some test written by repository holder
+8. Your code changes should withstand those tests
+9. If you pass test and have no conflict with master then it will show mergable
```
Comment (nathfreder) [inline how_to_contribute.md]: "If you pass test" should be "If you pass the test".

Comment (nathfreder) [reply how_to_contribute.md]: "mergable" is not a word.

File: how_to_contribute.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Steps to be followed to contribute to any open repository are :-
+
+1. Browse the Alex repository
+2. Have a look at the issues and if you are interested in any issue then clone the repository
+3. Always make your own branch on their repository
+4. Never make changes on master directly
+5. Do the necessary changes needed in the code to solve the problem
+6. push it on the github and as you push it will show you that branch is mergable or not
```
Comment (nathfreder) [inline how_to_contribute.md]: "mergable" is not a word.

**Comment by nathfreder** [comment]
Looks better, probably still needs work but it is a good start!

**Comment by wooorm** [comment]
Hey @bvishal8510, thanks for working on this! We already have a contributing file, and you can see it here: https://github.com/get-alex/alex/blob/master/contributing.md. I agree with @nathfreder that this is very specific, it could be a bit more lively and fun, more human.
Could you add a section to our existing file instead of this new duplicate?

**Comment by nathfreder** [comment]
@bvishal8510 I just got a PR for one on my project, it is well written and fun. If you would like me to modify and send in a PR, just ask. https://github.com/poggit/requests/blob/master/CONTRIBUTING.md

**Comment by wooorm** [comment]
Hey @bvishal8510, unfortunately someone else has now, after 3 days of no response, worked this out in GH-211, but still, thanks for working on this!

If you’re up for another challenge, [do peruse the other issues](https://github.com/get-alex/alex/issues).

**Pull request closed without merging**